src/lib/server/product.ts

```diff
@@ -155,3 +159,34 @@ export async function amountOfProductSold(productId: string): Promise<number> {
 		)?.total ?? 0
 	);
 }
+
+export function productPriceWithVariations(
+	product: Product,
+	chosenVariations: Record<string, string> | undefined
+) {
+	let variationPriceArray: Price[] = [];
+	const variationNamesInDB = [...new Set(product.variations?.map((vari) => vari.name))];
+	const chosenVariationNames = Object.keys(chosenVariations ?? {});
+
+	const allVariationsChosen =
+		variationNamesInDB.length === chosenVariationNames.length &&
+		variationNamesInDB.every((name) => chosenVariationNames.includes(name));
+
+	if (allVariationsChosen) {
+		variationPriceArray = chosenVariations
+			? Object.entries(chosenVariations).map((variation) => ({
+					amount:
+						product.variations?.find(
+							(vari) => variation[0] === vari.name && variation[1] === vari.value
+						)?.price ?? 0,
+					currency: product.price.currency
+			  }))
+			: [];
+	} else {
+		throw error(400, 'error matching on variations choice');
+	}
+
+	return product.hasVariations && chosenVariations
+		? sumCurrency(product.price.currency, [...variationPriceArray, product.price])
+		: 0;
```

Comment:
```suggestion
	return sumCurrency(product.price.currency, [...variationPriceArray, product.price]);
```

The check is useless or should have been done before this point.

(and we shouldn't return `0`)